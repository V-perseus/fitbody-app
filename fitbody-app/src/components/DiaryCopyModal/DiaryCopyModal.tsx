import React from 'react'
import { View, Text, TouchableOpacity } from 'react-native'
import Modal from 'react-native-modal'
import moment, { Moment } from 'moment'

// Components
import CloseIcon from '../../../assets/images/svg/icon/24px/close.svg'

// Assets
import styles from './styles'
import globals from '../../config/globals'

interface IDIaryCopyModal {
  sourceDate: Moment
  showModal: boolean
  closeModal: () => void
  selectDayHandler: (day: Moment) => void
  selectedDays: Moment[]
  createButtonHandler: () => void
}
const DiaryCopyModal: React.FC<IDIaryCopyModal> = ({
  sourceDate,
  showModal,
  closeModal,
  selectDayHandler,
  selectedDays,
  createButtonHandler,
}) => {
  // const mealPlans = useSelector((state) => state.data.meal.mealPlans)

  const renderWeek = (isThisWeek: boolean) => {
    const today = moment()
    const weekStart = today.clone().startOf('week')
    const weekEnd = today.clone().endOf('week')
    const days: JSX.Element[] = []

    for (let i = 0; i <= 6; i++) {
      const day = isThisWeek ? moment(weekStart).add(i, 'd') : moment(weekEnd).add(i + 1, 'd')

      const disabled = day.isSame(sourceDate, 'day')
      // || mealPlans[day.year()]?.[day.month()]?.[day.date()]?.meal_set

      days.push(
        <View key={i} style={styles.date}>
          <Text style={styles.weekDay}>{day.format('ddd').toUpperCase()}</Text>
          <TouchableOpacity
            style={[
              styles.dateWrapper,
              selectedDays.findIndex((d) => d.isSame(day)) >= 0 && styles.activeDate,
              disabled && styles.disabledDate,
            ]}
            disabled={disabled}
            onPress={() => selectDayHandler(day)}>
            <Text style={[styles.dateLabel, disabled && styles.disabledDateLabel]}>{day.format('D').toUpperCase()}</Text>
          </TouchableOpacity>
        </View>,
      )
    }
    return days
  }

  return (
    <Modal isVisible={showModal} backdropOpacity={0.5} onBackdropPress={closeModal} style={{ margin: 0 }}>
      <View style={styles.modalContainer}>
        <View style={styles.modalContents}>
          <TouchableOpacity onPress={closeModal} style={styles.closeButton} testID="close_btn">
            <CloseIcon color={globals.styles.colors.colorBlack} />
          </TouchableOpacity>
          <Text style={styles.modalTitle}>Copy Selected Foods</Text>
          <Text style={styles.modalDescription}>Choose which day(s) to{'\n'}copy selected foods to.</Text>
          <Text style={styles.weekHeader}>This Week</Text>
          <View style={styles.weekDays}>{renderWeek(true)}</View>
          <Text style={styles.weekHeader}>Next Week</Text>
          <View style={styles.weekDays}>{renderWeek(false)}</View>
          <TouchableOpacity style={styles.modalButton} onPress={createButtonHandler}>
            <Text style={styles.modalButtonText}>COPY</Text>
          </TouchableOpacity>
        </View>
      </View>
    </Modal>
  )
}

export default DiaryCopyModal
