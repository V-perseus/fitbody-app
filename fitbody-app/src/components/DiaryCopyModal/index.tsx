import DiaryCopyModal from './DiaryCopyModal'

export default DiaryCopyModal
