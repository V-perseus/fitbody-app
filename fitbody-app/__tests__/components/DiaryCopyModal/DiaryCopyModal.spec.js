import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react-native'
import renderer from 'react-test-renderer'
import moment from 'moment'

import DiaryCopyModal from '../../../src/components/DiaryCopyModal'

describe('DiaryCopyModal', () => {
  const createButtonHandler = jest.fn()
  const closeButton = jest.fn()

  afterEach(() => {
    cleanup()
  })

  it('renders snapshot correctly', () => {
    const tree = renderer
      .create(
        <DiaryCopyModal
          showModal={true}
          closeModal={closeButton}
          sourceDate={moment()}
          selectedDays={[]}
          selectDayHandler={() => {}}
          createButtonHandler={createButtonHandler}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('handles button presses', async () => {
    const { getByText, getByTestId } = render(
      <DiaryCopyModal
        showModal={true}
        closeModal={closeButton}
        sourceDate={moment()}
        selectedDays={[]}
        selectDayHandler={() => {}}
        createButtonHandler={createButtonHandler}
      />,
    )
    const copyBtn = getByText(/COPY/)
    expect(copyBtn).not.toBeNull()
    await fireEvent.press(copyBtn)
    expect(createButtonHandler).toBeCalledTimes(1)
    const closeBtn = getByTestId('close_btn')
    expect(closeBtn).not.toBeNull()
    await fireEvent.press(closeBtn)
    expect(closeButton).toBeCalledTimes(1)
  })

  it('renders all text', async () => {
    const { getByText } = render(
      <DiaryCopyModal
        showModal={true}
        closeModal={closeButton}
        sourceDate={moment()}
        selectedDays={[]}
        selectDayHandler={() => {}}
        createButtonHandler={createButtonHandler}
      />,
    )
    const titleText = getByText(/Copy Selected Foods/)
    const descriptionText = getByText(/Choose which day\(s\) to/i)
    const descriptionText2 = getByText(/copy selected foods to./)
    const thisWeekText = getByText(/This Week/)
    const nextWeekText = getByText(/Next Week/i)
    expect(titleText).not.toBeNull()
    expect(descriptionText).not.toBeNull()
    expect(descriptionText2).not.toBeNull()
    expect(thisWeekText).not.toBeNull()
    expect(nextWeekText).not.toBeNull()
  })
})
