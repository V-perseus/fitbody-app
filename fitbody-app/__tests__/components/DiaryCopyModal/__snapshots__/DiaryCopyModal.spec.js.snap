// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiaryCopyModal renders snapshot correctly 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[MockFunction]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  statusBarTranslucent={false}
  supportedOrientations={
    [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    collapsable={false}
    deviceHeight={null}
    deviceWidth={null}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[MockFunction]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    style={
      {
        "flex": 1,
        "justifyContent": "center",
        "margin": 0,
        "transform": [
          {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingHorizontal": 27,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 8,
            "flexDirection": "column",
            "height": 393,
            "paddingBottom": 29,
            "paddingHorizontal": 29,
            "paddingTop": 30,
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
              "position": "absolute",
              "right": 13,
              "top": 11,
            }
          }
          testID="close_btn"
        >
          <close.svg
            color="#333333"
          />
        </View>
        <Text
          style={
            {
              "color": "#000000",
              "fontFamily": "OpenSans-Bold",
              "fontSize": 17,
              "letterSpacing": -0.36,
              "textAlign": "center",
            }
          }
        >
          Copy Selected Foods
        </Text>
        <Text
          style={
            {
              "color": "#333333",
              "fontFamily": "OpenSans",
              "fontSize": 14,
              "marginBottom": 8,
              "marginTop": 5,
              "maxWidth": 276,
              "textAlign": "center",
            }
          }
        >
          Choose which day(s) to
          

          copy selected foods to.
        </Text>
        <Text
          style={
            {
              "color": "#bbbbbb",
              "fontFamily": "OpenSans-Semibold",
              "fontSize": 12,
              "marginBottom": 9,
              "marginTop": 18,
              "textAlign": "center",
            }
          }
        >
          This Week
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "column",
                "justifyContent": "center",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#bbbbbb",
                  "fontFamily": "OpenSans-Semibold",
                  "fontSize": 10,
                  "marginBottom": 2,
                  "textAlign": "center",
                }
              }
            >
              SUN
            </Text>
            <View
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#bbbbbb",
                  "borderRadius": 15,
                  "flexDirection": "row",
                  "height": 30,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 30,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#ffffff",
                      "fontFamily": "BebasNeue",
                      "fontSize": 14,
                      "textAlign": "center",
                    },
                    false,
                  ]
                }
              >
                2
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "column",
                "justifyContent": "center",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#bbbbbb",
                  "fontFamily": "OpenSans-Semibold",
                  "fontSize": 10,
                  "marginBottom": 2,
                  "textAlign": "center",
                }
              }
            >
              MON
            </Text>
            <View
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#bbbbbb",
                  "borderRadius": 15,
                  "flexDirection": "row",
                  "height": 30,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 30,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#ffffff",
                      "fontFamily": "BebasNeue",
                      "fontSize": 14,
                      "textAlign": "center",
                    },
                    false,
                  ]
                }
              >
                3
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "column",
                "justifyContent": "center",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#bbbbbb",
                  "fontFamily": "OpenSans-Semibold",
                  "fontSize": 10,
                  "marginBottom": 2,
                  "textAlign": "center",
                }
              }
            >
              TUE
            </Text>
            <View
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#bbbbbb",
                  "borderRadius": 15,
                  "flexDirection": "row",
                  "height": 30,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 30,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#ffffff",
                      "fontFamily": "BebasNeue",
                      "fontSize": 14,
                      "textAlign": "center",
                    },
                    false,
                  ]
                }
              >
                4
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "column",
                "justifyContent": "center",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#bbbbbb",
                  "fontFamily": "OpenSans-Semibold",
                  "fontSize": 10,
                  "marginBottom": 2,
                  "textAlign": "center",
                }
              }
            >
              WED
            </Text>
            <View
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#bbbbbb",
                  "borderRadius": 15,
                  "flexDirection": "row",
                  "height": 30,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 30,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#ffffff",
                      "fontFamily": "BebasNeue",
                      "fontSize": 14,
                      "textAlign": "center",
                    },
                    false,
                  ]
                }
              >
                5
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "column",
                "justifyContent": "center",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#bbbbbb",
                  "fontFamily": "OpenSans-Semibold",
                  "fontSize": 10,
                  "marginBottom": 2,
                  "textAlign": "center",
                }
              }
            >
              THU
            </Text>
            <View
              accessibilityState={
                {
                  "disabled": true,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#f3f3f3",
                  "borderRadius": 15,
                  "color": "#bbbbbb",
                  "flexDirection": "row",
                  "height": 30,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 30,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#ffffff",
                      "fontFamily": "BebasNeue",
                      "fontSize": 14,
                      "textAlign": "center",
                    },
                    {
                      "color": "#bbbbbb",
                    },
                  ]
                }
              >
                6
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "column",
                "justifyContent": "center",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#bbbbbb",
                  "fontFamily": "OpenSans-Semibold",
                  "fontSize": 10,
                  "marginBottom": 2,
                  "textAlign": "center",
                }
              }
            >
              FRI
            </Text>
            <View
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#bbbbbb",
                  "borderRadius": 15,
                  "flexDirection": "row",
                  "height": 30,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 30,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#ffffff",
                      "fontFamily": "BebasNeue",
                      "fontSize": 14,
                      "textAlign": "center",
                    },
                    false,
                  ]
                }
              >
                7
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "column",
                "justifyContent": "center",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#bbbbbb",
                  "fontFamily": "OpenSans-Semibold",
                  "fontSize": 10,
                  "marginBottom": 2,
                  "textAlign": "center",
                }
              }
            >
              SAT
            </Text>
            <View
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#bbbbbb",
                  "borderRadius": 15,
                  "flexDirection": "row",
                  "height": 30,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 30,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#ffffff",
                      "fontFamily": "BebasNeue",
                      "fontSize": 14,
                      "textAlign": "center",
                    },
                    false,
                  ]
                }
              >
                8
              </Text>
            </View>
          </View>
        </View>
        <Text
          style={
            {
              "color": "#bbbbbb",
              "fontFamily": "OpenSans-Semibold",
              "fontSize": 12,
              "marginBottom": 9,
              "marginTop": 18,
              "textAlign": "center",
            }
          }
        >
          Next Week
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "column",
                "justifyContent": "center",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#bbbbbb",
                  "fontFamily": "OpenSans-Semibold",
                  "fontSize": 10,
                  "marginBottom": 2,
                  "textAlign": "center",
                }
              }
            >
              SUN
            </Text>
            <View
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#bbbbbb",
                  "borderRadius": 15,
                  "flexDirection": "row",
                  "height": 30,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 30,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#ffffff",
                      "fontFamily": "BebasNeue",
                      "fontSize": 14,
                      "textAlign": "center",
                    },
                    false,
                  ]
                }
              >
                9
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "column",
                "justifyContent": "center",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#bbbbbb",
                  "fontFamily": "OpenSans-Semibold",
                  "fontSize": 10,
                  "marginBottom": 2,
                  "textAlign": "center",
                }
              }
            >
              MON
            </Text>
            <View
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#bbbbbb",
                  "borderRadius": 15,
                  "flexDirection": "row",
                  "height": 30,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 30,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#ffffff",
                      "fontFamily": "BebasNeue",
                      "fontSize": 14,
                      "textAlign": "center",
                    },
                    false,
                  ]
                }
              >
                10
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "column",
                "justifyContent": "center",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#bbbbbb",
                  "fontFamily": "OpenSans-Semibold",
                  "fontSize": 10,
                  "marginBottom": 2,
                  "textAlign": "center",
                }
              }
            >
              TUE
            </Text>
            <View
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#bbbbbb",
                  "borderRadius": 15,
                  "flexDirection": "row",
                  "height": 30,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 30,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#ffffff",
                      "fontFamily": "BebasNeue",
                      "fontSize": 14,
                      "textAlign": "center",
                    },
                    false,
                  ]
                }
              >
                11
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "column",
                "justifyContent": "center",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#bbbbbb",
                  "fontFamily": "OpenSans-Semibold",
                  "fontSize": 10,
                  "marginBottom": 2,
                  "textAlign": "center",
                }
              }
            >
              WED
            </Text>
            <View
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#bbbbbb",
                  "borderRadius": 15,
                  "flexDirection": "row",
                  "height": 30,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 30,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#ffffff",
                      "fontFamily": "BebasNeue",
                      "fontSize": 14,
                      "textAlign": "center",
                    },
                    false,
                  ]
                }
              >
                12
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "column",
                "justifyContent": "center",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#bbbbbb",
                  "fontFamily": "OpenSans-Semibold",
                  "fontSize": 10,
                  "marginBottom": 2,
                  "textAlign": "center",
                }
              }
            >
              THU
            </Text>
            <View
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#bbbbbb",
                  "borderRadius": 15,
                  "flexDirection": "row",
                  "height": 30,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 30,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#ffffff",
                      "fontFamily": "BebasNeue",
                      "fontSize": 14,
                      "textAlign": "center",
                    },
                    false,
                  ]
                }
              >
                13
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "column",
                "justifyContent": "center",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#bbbbbb",
                  "fontFamily": "OpenSans-Semibold",
                  "fontSize": 10,
                  "marginBottom": 2,
                  "textAlign": "center",
                }
              }
            >
              FRI
            </Text>
            <View
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#bbbbbb",
                  "borderRadius": 15,
                  "flexDirection": "row",
                  "height": 30,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 30,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#ffffff",
                      "fontFamily": "BebasNeue",
                      "fontSize": 14,
                      "textAlign": "center",
                    },
                    false,
                  ]
                }
              >
                14
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "column",
                "justifyContent": "center",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#bbbbbb",
                  "fontFamily": "OpenSans-Semibold",
                  "fontSize": 10,
                  "marginBottom": 2,
                  "textAlign": "center",
                }
              }
            >
              SAT
            </Text>
            <View
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#bbbbbb",
                  "borderRadius": 15,
                  "flexDirection": "row",
                  "height": 30,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 30,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#ffffff",
                      "fontFamily": "BebasNeue",
                      "fontSize": 14,
                      "textAlign": "center",
                    },
                    false,
                  ]
                }
              >
                15
              </Text>
            </View>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fa8dd5",
              "borderRadius": 8,
              "flexDirection": "column",
              "height": 55,
              "justifyContent": "center",
              "marginTop": 30,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              {
                "color": "#ffffff",
                "fontFamily": "OpenSans-Semibold",
                "fontSize": 16,
                "letterSpacing": -0.39,
                "lineHeight": 22,
                "textAlign": "center",
              }
            }
          >
            COPY
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
